File: TWO-SUM/ts.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hm = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (hm.containsKey(diff)) {
                return new int[]{hm.get(diff), i};
            }
            hm.put(nums[i], i);
        }
        return null;
        }

    public static void main(String[] args) {
        Solution s1 = new Solution();
        int[] nums = {2, 7, 11, 15};
        int target = 9;
        int[] res = s1.twoSum(nums, target);
}

File: TWO-SUM/ts.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // int tempIndex = 0;
        // for(int i = 0; i < nums.size(); i++){
        //     tempIndex = i;
        //     for (int j = i + 1; j< nums.size(); j++){
        //         if(nums[i] + nums[j] == target){
        //             return vector<int>{i,j};
        //         }
        //     }
        // }
        // return vector<int>{};
        unordered_map<int, int> hm;
        for (int i = 0; i < nums.size(); i++) {
            int diff = target - nums[i];
            if (hm.find(diff) != hm.end()) {
                return vector<int>{i, hm[diff]};
            }
            hm[nums[i]] = i;
        }
        return {};
    }
};

int main() {
    Solution s;
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;
    vector<int> result = s.twoSum(nums, target);
    for(int i : result) {
        cout << i << " ";
    }
    cout << endl;  // Output: 0 1
    return 0;
}